ui/components/app/confirmation-warning-modal/confirmation-warning-modal.js

```diff
@@ -2,102 +2,119 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { useI18nContext } from '../../../hooks/useI18nContext';
 
-import Popover from '../../ui/popover';
-import Box from '../../ui/box';
-import Button from '../../ui/button';
+import { Box } from '../../component-library/box';
 import {
   Display,
   FlexDirection,
   FontWeight,
-  JustifyContent,
   TextVariant,
   AlignItems,
   IconColor,
 } from '../../../helpers/constants/design-system';
-import { Icon, IconName, IconSize, Text } from '../../component-library';
+import {
+  Icon,
+  IconName,
+  IconSize,
+  Text,
+  Modal,
+  ModalContent,
+  ModalOverlay,
+  Button,
+  BUTTON_VARIANT,
+} from '../../component-library';
 
 const ConfirmationWarningModal = ({ onSubmit, onCancel }) => {
   const t = useI18nContext();
 
   return (
-    <Popover
+    <Modal
+      isOpen
+      onClose={onCancel}
       className="confirmation-warning-modal__content"
-      footer={
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <Box
+          display={Display.Flex}
+          flexDirection={FlexDirection.Row}
+          alignItems={AlignItems.center}
+          padding={3}
+          margin={0}
+          className="confirmation-warning-modal__content__header"
+        >
+          <Icon
+            name={IconName.Danger}
+            color={IconColor.errorDefault}
+            className="confirmation-warning-modal__content__header__warning-icon"
+            size={IconSize.Xl}
+          />
+          <Text
+            variant={TextVariant.headingSm}
+            as="h4"
+            fontWeight={FontWeight.Bold}
+          >
+            {t('addEthereumChainWarningModalTitle')}
+          </Text>
+        </Box>
```

Comment:
Could we use `ModalHeader` here with this example using `children` and `childrenWrapperProps`

```
<ModalHeader
  childrenWrapperProps={{
    display: Display.Flex,
    flexDirection: FlexDirection.Column,
    alignItems: AlignItems.center,
    justifyContent: JustifyContent.center,
   gap: 4
  }}
>
  <Icon
    name={IconName.Danger}
    color={IconColor.errorDefault}
    className="confirmation-warning-modal__content__header__warning-icon"
    size={IconSize.Xl}
  />
  <Text variant={TextVariant.headingSm} textAlign={TextAlign.Center}>
    Custom header using multiple components
  </Text>
</ModalHeader>
```